astro/src/content/quickstarts/quickstart-react-native.mdx

```diff
@@ -0,0 +1,208 @@
+            lang="ini"/>
+
+Edit `app.json` to add a `scheme` that will be used when redirecting back to your application after logging in and out of FusionAuth.
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-react-native/main/complete-application/app.json"
+            lang="json"/>
+
+## App Customization
+
+In this section, you'll turn your application into a trivial banking application with some styling.
+
+### Add Styling
+
+First, run the command below to install some libraries needed for theming.
+
+```shell
+npm install expo-image expo-constants react-native-currency-input
+```
+
+Instead of using CSS, React Native has its own concept of [stylesheets](https://reactnative.dev/docs/stylesheet). Create a file named `changebank.style.js` with the contents below to style your ChangeBank app.
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-react-native/main/complete-application/changebank.style.js"
+            lang="javascript"/>
+
+## Finish Setting up the App
+
+Change the existing `App.js` to integrate `expo-auth-session`, add the ChangeBank template and stitch up everything.
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-react-native/main/complete-application/App.js"
+            lang="javascript"/>
+
+## Run the App
+
+If you want to test on an Android device, connect a physical device via USB or install [Android Studio](https://developer.android.com/studio) to set up an emulator.
+
+If you want to test on an iOS device and are running a Mac, install [Xcode](https://developer.apple.com/xcode/) to set up an emulator.
+Start up the React Native application using this command:
+
+```shell
+npx expo start
+```
+
+After waiting a few moments, you should see a QR Code and a menu with some actions. Right below the QR Code, you'll see a message like this one _(the real address may vary)_.
+
+```
+› Metro waiting on exp://192.168.1.2:8081
+```
+
+To use [Expo Go](https://docs.expo.dev/get-started/expo-go/), a client for testing your apps on Android and iOS devices without building anything locally, you need to:
+
+* Copy that listening address (`exp://192.168.1.2:8081`).
+* Navigate to your FusionAuth instance on [localhost:9011](http://localhost:9011).
+* Browse to **Applications**.
+* Click on <IconButton icon="edit" color="blue" /> in your `ExampleReactNativeApp` to edit it.
+* Go to the **OAuth** tab.
+* Add that address to **Authorized redirect URLs** and append `/--/redirect` to it
+  * Example: for `exp://192.168.1.2:8081`, you'd need to add `exp://192.168.1.2:8081/--/redirect` as a redirect URL
+* Click on <IconButton icon="save" color="blue" /> to save your changes.
+
+Go back to the terminal with the Expo menu. To test on the connected or emulated Android device, press `a`. Otherwise, press `i` to run on the emulated iOS device.
```

Comment:
I thought it may have to do with the first time you use the ngrok address it asked if you wanted to allow.  I used safari and accepted that, but still does not work.

Reply:
I kept fiddling and think I got it to work.  There was something weird with the way ngrok started up.  It appears to run and I can get to the fusionauth server, now I am getting a client authentication failed.
![Screenshot 2023-09-05 at 4 13 37 PM](https://github.com/FusionAuth/fusionauth-site/assets/137117976/ce144ee4-4392-4939-8347-6402155f1507)

Reply:
Need to test logout too.

Reply:
So now that I have the proper kickstart running, the application works.  I would be curious to know what I need to add to an existing instance of FusionAuth to get it to run.  I noticed the io.fusionauth.app://redirect as an Authorized redirect URL and that may be it.  You may want to make a note of that being in the FusionAuth settings and why.

Reply:
Hmm of course I can do that, but I don't think we've been doing that to other quickstarts, right? I mean, how to set up existing instances.

I do explain what needs to be done in a few paragraphs below (in _"To use Expo Go, a client for testing your apps on Android and iOS devices without building anything locally, you need to:"_).

Reply:
You are right that we don't do that with other quickstarts so you are probably good there.  

Now that I read through the instructions again, it does seem to make more sense.  It maybe that I was just not familiar with the platform and that is why I struggled.